aioredis/stream.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+
+
+class StreamReader(asyncio.StreamReader):
+    """
+    Override the official StreamReader to address the
+    following issue: http://bugs.python.org/issue30861
+
+    Also it leverages to get rid of the dobule buffer and
+    get rid of one coroutine step. Data flows from the buffer
+    to the Redis parser directly.
+    """
+
+    def set_parser(self, parser):
+        self._parser = parser
+
+    def feed_data(self, data):
+        assert not self._eof, 'feed_data after feed_eof'
+
+        if not data:
+            return
+
+        self._parser.feed(data)
+        self._wakeup_waiter()
+
+        # TODO: implement pause the read. Its needed
+        #       expose the len of the buffer from hiredis
+        #       to make it possible.
```

Comment:
We don't have visibility, at least for the `hiredis` version, of the `len` of the buffer. My plans will pass doing the proper MR to expose this property by the `hiredis-py` project, doesn't look a rocket science.  With that attribute, we can implement the proper pause/resume option.